### PR TITLE
meta: auto-lock closed issues/PRs

### PR DESCRIPTION
**Pull request by chadwhitacre**
https://app.asana.com/0/1199602781063150/1198192573081731

This locks all closed issues and PRs that have not been updated (e.g., commented on) for at least 15 days, via a third-party action initially vetted in https://github.com/getsentry/onpremise/pull/772 and also applied in https://github.com/getsentry/sentry-docs/pull/2780. It's set to run as frequently as possible in order to work down the initial backlog, which is about 22,000 issues/PRs. On `sentry-docs` I'm seeing real-world throughput of ~190/hr, which puts us at four or five days to catch up here on `sentry`. Once we do, I'll follow up with a second PR to back off to twice a day (should be plenty to keep up with the ~25/day we're seeing currently).

### CODE REVIEW

File: .github/workflows/lock.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: 'lock closed issues/PRs'
+on:
+  schedule:
+    - cron: '*/2 * * * *'
+jobs:
+  lock:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dessant/lock-threads@v2
```
Comment (billyvg) [inline .github/workflows/lock.yml:9]: Let's fork this, `v2` here can be a tag or branch, both which are mutable, so even if the action is legit now, they can always change it from under us.

I plan on using GH settings to enforce valid sources of actions in the future (I'm unsure if these can be applied org wide or not).

**Review by billyvg — CHANGES_REQUESTED** [review]

File: .github/workflows/lock.yml (hunk 1 above)
Comment (chadwhitacre) [reply .github/workflows/lock.yml:9]: Forked to https://github.com/getsentry/lock-threads. Public or private?

Comment (chadwhitacre) [reply .github/workflows/lock.yml:9]: I'm going to test the fork on a smaller repo and then I'll return here.

Comment (billyvg) [reply .github/workflows/lock.yml:9]: public is fine

Comment (BYK) [reply .github/workflows/lock.yml:9]: No need for a fork, using a SHA should be enough to mitigate this as they are immutable

Comment (BYK) [reply .github/workflows/lock.yml:9]: I'm against forking every third-party thing we use if there are alternatives to that approach.

Comment (chadwhitacre) [reply .github/workflows/lock.yml]: Pinned via SHA in fb3e5cdac1d38459a598cdc3bc1799870ac6b4d5. You good with that, @billyvg?

**Review by billyvg — APPROVED** [review]
@chadwhitacre pinning to a sha is good but a fork would be better because we will be allowing all actions under getsentry org. https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-enterprise/enforcing-github-actions-policies-in-your-enterprise-account

**Comment by BYK** [comment]
>because we will be allowing all actions under getsentry org. 

Let's reevaluate this.

**Review by BYK — APPROVED** [review]

**Pull request merged**